typescript/graphql-nextjs/pages/drafts.tsx

```diff
@@ -18,28 +19,35 @@ const DraftsQuery = gql`
   }
 `
 
-const Post = ({ post }) => (
-  <Link href="/p/[id]" as={`/p/${post.id}`}>
-    <a>
-      <h2>{post.title}</h2>
-      <small>By {post.author ? post.author.name : "Unknown Author"}</small>
-      <p>{post.content}</p>
-      <style jsx>{`
-        a {
-          text-decoration: none;
-          color: inherit;
-          padding: 2rem;
-          display: block;
-        }
-      `}</style>
-    </a>
-  </Link>
-)
+type Drafts = DraftsQuery["drafts"]
+type Draft = NonNullable<Drafts>[number]
```

Comment:
The generated GraphQL types have the correct types you can use when making queries and mutations, let's use that here.

Refer to [Apollo Client 3 with GraphQL Code Generator](https://youtu.be/PYDGjTufGsk) video by Jamie Barton or the repository he created https://github.com/notrab/graphql.wtf-graphql-codegen-apollo-client on how to use the generated types in the application.

Reply:
The same applies to the sections where types for queries have been manually defined - let's use the generated types.